API/RestMatch.API/RestMatch.API/Controllers/RestaurantsController.cs

```diff
@@ -0,0 +1,191 @@
+using Microsoft.AspNetCore.Mvc;
+using RestMatch.API.Domain.Models;
+using RestMatch.API.Application.Dtos;
+using RestMatch.API.Application.Interfaces;
+
+namespace RestMatch.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class RestaurantsController : ControllerBase
+    {
+        private readonly IRestaurantService _service;
+
+        public RestaurantsController(IRestaurantService service)
+        {
+            _service = service;
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public async Task<ActionResult<ICollection<GetRestaurantResponseDto>>> GetRestaurants()
+        {
+            var restaurants = await _service.GetRestaurants();
+            if (restaurants.Count == 0)
+                return NoContent();
+
+            return Ok(restaurants);
+        }
+
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<GetRestaurantResponseDto>> GetRestaurant(int id)
+        {
+            var restaurant = await _service.GetRestaurant(id);
+            if (restaurant == null)
+                return NotFound($"{nameof(Restaurant)} wasn't found");
+
+            return Ok(restaurant);
+        }
+
+        [HttpGet("imageUrls/{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<GetRestaurantImageUrlResponseDto>> GetRestaurantImageUrl(int id)
+        {
+            var imageUrl = await _service.GetRestaurantImageUrl(id);
+            if (imageUrl == null)
+                return NotFound($"{nameof(RestaurantImageUrl)} wasn't found");
+
+            return Ok(imageUrl);
+        }
+
+        [HttpGet("{restaurantId}/imageUrls")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<ICollection<GetRestaurantImageUrlResponseDto>>> GetRestaurantImageUrls(int restaurantId)
```

Comment:
i suppose we don't need separate requests for restaurant, restaurant images and other stuff. The point was to retrieve it in one dto

Reply:
added